coretex/entities/task_run/utils.py

```diff
@@ -58,4 +64,48 @@ def fetchDataset(datasetType: Type[Dataset], value: Any) -> Optional[Dataset]:
     if issubclass(datasetType, NetworkDataset):
         return datasetType.fetchById(value)
 
-    return None
\ No newline at end of file
+    return None
+
+
+def getSnapshotFiles(dirPath: Path, ignoredFiles: List[str]) -> List[Path]:
+    snapshotFiles: List[Path] = []
+    if dirPath.joinpath(".coretexignore").exists():
+        return []
+
+    for path in dirPath.iterdir():
+        if path.is_dir():
+            snapshotFiles.extend(getSnapshotFiles(path, ignoredFiles))
+        elif str(path) not in ignoredFiles:
+            snapshotFiles.append(path)
+
+    return snapshotFiles
+
+
+def getDefaultEntryPoint() -> Optional[str]:
+    for defaultEntryPoint in [Path("./main.py"), Path("./main.r"), Path("./main.R")]:
+        if defaultEntryPoint.exists():
+            return defaultEntryPoint.name
+
+    return None
+
+
+def createSnapshot(entryPoint: Optional[str]) -> Path:
+    if entryPoint is None:
+        entryPoint = getDefaultEntryPoint()
+
+    if entryPoint is None or not Path(".", entryPoint).exists():
+        raise FileNotFoundError(">> [Coretex] Entry point file not found")
+
+    snapshotPath = folder_manager.temp / "snapshot.zip"
+    with ZipFile(snapshotPath, "w", ZIP_DEFLATED) as snapshotArchive:
+        repo = git.Repo("./", search_parent_directories = True)
+        ignoredFiles = repo.ignored(list(Path("./").rglob("*")))  # type: ignore
+
+        for requiredFile in [entryPoint, "requirements.txt"]:
```

Comment:
let's remove this `for` loop and check properly if `entryPoint` and `"requirements.txt"` exist and if not raise a `FileNotFound` error